### PR TITLE
Remove the Text2dSize component

### PR DESCRIPTION
**Pull request by ickshonpe**
# Objective

The Text2dSize component is redundant, the size of the text is already stored in TextLayoutInfo.

## Solution

Delete Text2dSize 

## Changelog

 changes:
        *  make Text2dBounds non-optional
        * added `With<Text2dBounds>` query filter to `extract_text2d_sprite`
        * removed the Text2dSize component
        * clarified the text2d alignment offset calculations
        * removes some unnecessary scale factor multiplications

### CODE REVIEW

File: crates/bevy_text/src/text2d.rs [hunk 1]
```diff
@@ -168,17 +160,14 @@ pub fn update_text2d_layout(
         Changed<Text>,
         &Text,
         Option<&Text2dBounds>,
-        &mut Text2dSize,
         Option<&mut TextLayoutInfo>,
```
Comment (devil-ira) [inline crates/bevy_text/src/text2d.rs:163]: This query now matches the non `Text2d` texts and breaks their layout.

Comment (ickshonpe) [reply crates/bevy_text/src/text2d.rs:163]: Ah didn't consider that.
A `Without<Node>` filter should do it?

**Comment by rparrett** [comment]
Did you consider making the size field on Text2dBounds optional instead?

This could use a `Migration Guide` entry for the `Text2dSize` removal.

And just so nobody else has to go digging: The duplicated layout info data was introduced in https://github.com/bevyengine/bevy/pull/4460.

**Comment by ickshonpe** [comment]
> Did you consider making the size field on Text2dBounds optional instead?

Text2dBound size's dimensions default to f32::MAX already, so it doesn't seem necessary.

**Comment by ickshonpe** [comment]
The issues here were fixed in #6807 with a better implementation, so closing this one.

**Pull request closed without merging**